### PR TITLE
improvement(re-ordering of components)

### PR DESCRIPTION
**Pull request by ashwinKumar0505**
The components can now be reordered using drag and drop.

### CODE REVIEW

**Comment by ashwinKumar0505** [comment]
The dragged component will be re-ordered only when it crosses half of the next component present to it. As the next component can be present on the right or left and also top or bottom. We have to check both the x-axis and y-axis of the next component while dragging. thus while dragging from top to bottom, Not only y-axis middle is checked but also x-axis is checked. Same if we drag component from left to right, Not only x-axis is checked but also y-axis is checked.This leads to contradiction

File: src/hooks/useInteractive.ts [hunk 1]
```diff
@@ -7,6 +7,7 @@ import {
   getIsHovered,
 } from '../core/selectors/components'
 import { getShowLayout, getFocusedComponent } from '../core/selectors/app'
+import { useDropComponent } from './useDropComponent'
```
Comment (sanandnarayan) [inline src/hooks/useInteractive.ts]: not sure if this is a good idea. useInteractive is a custom hook which makes a component draggable. Not sure if adding useDropComponent hook here is a good idea

File: src/hooks/useDropComponent.ts [hunk 2]
```diff
@@ -1,26 +1,66 @@
...
+        const fromIndex = components[selectedComponent.parent].children.indexOf(
+          item.id,
+        )
+        const toIndex = components[selectedComponent.parent].children.indexOf(
+          componentId,
+        )
+        const hoverClientY = clientOffset && clientOffset.y - top
+
+        // Dragging downwards
+        if (fromIndex < toIndex && hoverClientY && hoverClientY < hoverMiddleY)
```
Comment (sanandnarayan) [inline src/hooks/useDropComponent.ts]: Need better comments. 
`hasPassedMid?`

**Review by sanandnarayan — CHANGES_REQUESTED** [review]
You might want to rethink the implementation. 
can we get inspiration from 
`https://react-dnd.github.io/react-dnd/examples/sortable/simple`

**Comment by ashwinKumar0505** [comment]
> You might want to rethink the implementation.
> can we get inspiration from
> `https://react-dnd.github.io/react-dnd/examples/sortable/simple`

This is implemented with the help of that only bro.

File: src/hooks/useInteractive.ts (hunk 1 above)
Comment (ashwinKumar0505) [reply src/hooks/useInteractive.ts]: As useInteractive is a common hook , I used there to reduce the lines of code and also we need ref to know the boundary points of the component.  we are creating ref in the useInteractive. These are the two reasons why i used useDropComponent in useInteractive. I will find a way to use it outside bro

**Pull request closed without merging**